frc46_token/src/token/state.rs

```diff
@@ -523,7 +530,8 @@ impl TokenState {
         let res = balances.for_each(|owner_key, balance| {
             let owner = match decode_actor_id(owner_key) {
                 None => {
-                    bail!("invalid key in balances map")
+                    maybe_err = Some(StateInvariantError::InvalidOwnerKey(owner_key.clone()));
+                    bail!("invariant failed");
```

Comment:
I should have caught this earlier, let's fix this while we're here.

The general idea behind these checks in the builtin actors is to always proceed, but accumulate errors to be returned. The methods in builtin actors don't even return a Result (but they do unwrap() some impossible-to-recover things). They basically return a list of strings describing any problems encountered, and the state summary. [Example](https://github.com/filecoin-project/builtin-actors/blob/master/actors/power/src/testing.rs#L36-L40).

Could we update to a similar pattern here? You can't depend on the MessageAccumulator type, but just returning a vector of strings will be fine.

Also then avoid using `anyhow` for `bail`.